warehouse/utils/utils_test.go

```diff
@@ -1,139 +1,917 @@
-
-					location, region = GetS3Location("https://test-bucket.s3.any-region.amazonaws.com/test-object.csv")
-					Expect(region).To(Equal("any-region"))
-					Expect(location).To(Equal("s3://test-bucket/test-object.csv"))
-
-					location, region = GetS3Location("https://my.test-bucket.s3.amazonaws.com/test-object.csv")
-					Expect(region).To(Equal(""))
-					Expect(location).To(Equal("s3://my.test-bucket/test-object.csv"))
-
-					location, region = GetS3Location("https://my.test-bucket.s3.us-west-1.amazonaws.com/test-object.csv")
-					Expect(region).To(Equal("us-west-1"))
-					Expect(location).To(Equal("s3://my.test-bucket/test-object.csv"))
-				})
-			})
-
-			Context("GetS3LocationFolder", func() {
-				It("should parse url and return location folder", func() {
-					var location string
-
-					location = GetS3LocationFolder("https://test-bucket.s3.amazonaws.com/myfolder/test-object.csv")
-					Expect(location).To(Equal("s3://test-bucket/myfolder"))
-
-					location = GetS3LocationFolder("https://test-bucket.s3.eu-west-2.amazonaws.com/myfolder/test-object.csv")
-					Expect(location).To(Equal("s3://test-bucket/myfolder"))
-
-					location = GetS3LocationFolder("https://my.test-bucket.s3.eu-west-2.amazonaws.com/myfolder/test-object.csv")
-					Expect(location).To(Equal("s3://my.test-bucket/myfolder"))
-				})
-			})
-
-			Context("GetS3Locations", func() {
-				It("should parse multiple urls and return array with locations ", func() {
-					inputs := []LoadFileT{
-						{Location: "https://test-bucket.s3.amazonaws.com/test-object.csv"},
-						{Location: "https://test-bucket.s3.eu-west-1.amazonaws.com/test-object.csv"},
-						{Location: "https://my.test-bucket.s3.amazonaws.com/test-object.csv"},
-						{Location: "https://my.test-bucket.s3.us-west-1.amazonaws.com/test-object.csv"},
-					}
-
-					locations := GetS3Locations(inputs)
-					Expect(locations[0].Location).To(Equal("s3://test-bucket/test-object.csv"))
-					Expect(locations[1].Location).To(Equal("s3://test-bucket/test-object.csv"))
-					Expect(locations[2].Location).To(Equal("s3://my.test-bucket/test-object.csv"))
-					Expect(locations[3].Location).To(Equal("s3://my.test-bucket/test-object.csv"))
-				})
-			})
+func TestGetS3Location(t *testing.T) {
+	inputs := []struct {
+		location   string
+		s3Location string
+		region     string
+	}{
+		{
+			location:   "https://test-bucket.s3.amazonaws.com/test-object.csv",
+			s3Location: "s3://test-bucket/test-object.csv",
+			region:     "",
+		},
+		{
+			location:   "https://test-bucket.s3.any-region.amazonaws.com/test-object.csv",
+			s3Location: "s3://test-bucket/test-object.csv",
+			region:     "any-region",
+		},
+		{
+			location:   "https://my.test-bucket.s3.amazonaws.com/test-object.csv",
+			s3Location: "s3://my.test-bucket/test-object.csv",
+			region:     "",
+		},
+		{
+			location:   "https://my.test-bucket.s3.us-west-1.amazonaws.com/test-object.csv",
```

Comment:
what happens when bucket name is `my.example.s3.bucket`

Reply:
Also added in the test suite as well. Thanks @Jayachand for the feedback.